docs/mp/health/01_introduction.adoc

```diff
@@ -114,24 +141,364 @@ microservice reports itself as ready.
+
+For example, you can specify a custom port and root context for the root health endpoint path.
+However, you cannot use different ports, such as http://localhost:8080/myhealth and http://localhost:8081/myhealth/live.
+Likewise, you cannot use different paths, such as http://localhost:8080/health and http://localhost:8080/probe/live.
+The example below will change the root path.
+[source,properties]
+.Create a file named `microprofile-config.properties` in the `resources\META-INF` directory with the following contents:
+----
+health.web-context=myhealth  #<1>
+----
+<1> The web-context specifies a new root path for the health endpoint.
+
+== Examples
+
+Generate Helidon MP Quickstart project following these <<Instructions, Instructions>>
+
+=== Using the Built-In Health Checks
+
+Helidon has a set of built-in health checks that are enabled to report various
+health check statuses that are commonly used:
+
+* deadlock detection
+* available disk space
+* available heap memory
+
+The following example will demonstrate how to use the built-in health checks.  These examples are all executed
+from the root directory of your project (helidon-quickstart-mp).
+
+[source,bash]
+.Build the application, then run it:
+----
+mvn package
+java -jar target/helidon-quickstart-mp.jar
+----
+
+[source,bash]
+.Verify the health endpoint in a new terminal window:
+----
+curl http://localhost:8080/health
+----
+
+[source,json]
+.JSON response:
+----
+{
+  "status": "UP",
+  "checks": [
+    {
+      "name": "deadlock",
+      "status": "UP"
+    },
+    {
+      "name": "diskSpace",
+      "status": "UP",
+      "data": {
+        "free": "325.54 GB",
+        "freeBytes": 349543358464,
+        "percentFree": "69.91%",
+        "total": "465.63 GB",
+        "totalBytes": 499963174912
+      }
+    },
+    {
+      "name": "heapMemory",
+      "status": "UP",
+      "data": {
+        "free": "230.87 MB",
+        "freeBytes": 242085696,
+        "max": "3.56 GB",
+        "maxBytes": 3817865216,
+        "percentFree": "98.90%",
+        "total": "271.00 MB",
+        "totalBytes": 284164096
+      }
+    }
+  ]
+}
+----
+
+=== Custom Liveness Health Checks
+
+You can create application-specific custom health checks and integrate them with Helidon
+using CDI.  The following example shows how to add a custom liveness health check.
+
+[source,java]
+.Create a new `GreetLivenessCheck` class with the following content:
+----
+@Liveness // <1>
+@ApplicationScoped // <2>
+public class GreetLivenessCheck implements HealthCheck {
+  private GreetingProvider provider;
+
+  @Override
+  public HealthCheckResponse call() {
+    return HealthCheckResponse.named("LivenessCheck")  // <3>
+        .up()
+        .withData("time", System.currentTimeMillis())
+        .build();
+  }
+}
+----
+<1> Annotation indicating this is a liveness health check.
+<2> Annotation indicating there is a single liveness `HealthCheck` object during the lifetime of the application.
```

Comment:
Annotation indicating this is a bean instantiated once per application (in Helidon this means just once per runtime)

Reply:
Replaced with comment above.